apollo-composition/src/lib.rs

```diff
@@ -355,3 +356,58 @@ fn satisfiability_result_into_issues(
         Err(issue) => Either::Right(once(issue)),
     }
 }
+
+impl From<BuildError> for Issue {
+    fn from(error: BuildError) -> Issue {
+        Issue {
+            code: error.code.expect("BuildError.code unexpectedly missing"),
+            message: error
+                .message
+                .expect("BuildError.message unexpectedly missing"),
+            locations: error
+                .nodes
+                .expect("BuildError.nodes unexpectedly missing")
+                .into_iter()
+                .map(Into::into)
+                .collect(),
+            severity: Severity::Error,
+        }
+    }
+}
+
+impl From<BuildHint> for Issue {
+    fn from(hint: BuildHint) -> Issue {
+        Issue {
+            code: hint.code.expect("BuildHint.code unexpectedly missing"),
+            message: hint.message,
+            locations: hint
+                .nodes
+                .expect("BuildHint.nodes unexpectedly missing")
+                .into_iter()
+                .map(Into::into)
+                .collect(),
+            severity: Severity::Warning,
+        }
+    }
+}
+
+impl From<BuildMessageLocation> for SubgraphLocation {
+    fn from(location: BuildMessageLocation) -> SubgraphLocation {
+        let start = location.start.unwrap_or_default();
+        let end = location.end.unwrap_or_default();
+
+        SubgraphLocation {
+            subgraph: location.subgraph.unwrap_or_default(),
+            range: Range {
+                start: LineColumn {
+                    line: start.line.unwrap_or(0),
+                    column: start.column.unwrap_or(0),
+                },
+                end: LineColumn {
+                    line: end.line.unwrap_or(0),
+                    column: end.column.unwrap_or(0),
+                },
+            },
+        }
```

Comment:
This should return `Option<SubgraphLocation>` and use `?` a bunch instead of defaults. See the way `from_ast` works.

Reply:
I went ahead and made this change. Is it weird to filter out locations where the linecol info is missing (as opposed to defaulting to 0,0)?

Reply:
Well setting the location to an arbitrary number doesn't feel right (it'd be 1, 1, not 0, 0)—we should semantically encode that there is no location with `None`. I _could_ see making the entire `range` optional, though, so we can keep subgraph name if it's set.

Actually, I think it doesn't make sense to produce locations _without_ subgraph names, so we should probably filter out any that are missing it.